### PR TITLE
Fix superlinter on the master branch and ensure it runs on PRs

### PR DESCRIPTION
**Pull request by jeancochrane**
Our superlinter checks [are failing on the master branch](https://github.com/ccao-data/data-architecture/actions/runs/5684699106/job/15407982040) but they [passed on the most recent PR](https://github.com/ccao-data/data-architecture/actions/runs/5684620829/job/15407722505) (#23) because the config points to the wrong name for our main branch:

https://github.com/ccao-data/data-architecture/blob/72bf150a8eb4117191f4a84053527c60ae5f8818/.github/workflows/lint.yaml#L30

``` 
2023-07-27 19:23:43 [INFO]   Gathering user validation information...
2023-07-27 19:23:43 [INFO]   - Only validating [new], or [edited] files in code base...
You are not currently on a branch.
Please specify which branch you want to merge with.
See git-pull(1) for details.

    git pull <remote> <branch>

fatal: ambiguous argument 'main...06569ff79d05062bf89c03297192d23ee331442c': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
2023-07-27 19:23:[49](https://github.com/ccao-data/data-architecture/actions/runs/5684620829/job/15407722505#step:4:50) [WARN]   No files were found in the GITHUB_WORKSPACE:[/github/workspace] to lint!
```

This PR points superlinter to the correct main branch name (`master`) and tweaks some files/configs to ensure it will pass on the next merge.

You can confirm that superlinter is in fact running the tests by checking out the [latest run for this PR](https://github.com/ccao-data/data-architecture/actions/runs/5685167543/job/15409465635).

### CODE REVIEW

File: .jscpd.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "ignore": ["dbt/models/**/*.sql"]
```
Comment (jeancochrane) [inline .jscpd.json:2]: These models necessarily involve duplication for now, so I'm ignoring them in the copy/paste linter.

File: .yamllint.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+rules:
+  document-start: disable
```
Comment (jeancochrane) [inline .yamllint.yml:2]: The document start marker (`---`) is not necessary in dbt files; the linter only emits this as a warning by default, but I found it annoying so I'm disabling it.

File: .github/workflows/lint.yaml [hunk 3]
```diff
@@ -27,6 +27,9 @@ jobs:
         uses: github/super-linter@v4
         env:
           VALIDATE_ALL_CODEBASE: false
-          DEFAULT_BRANCH: main
+          DEFAULT_BRANCH: master
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           VALIDATE_SQLFLUFF: false
+          VALIDATE_MARKDOWN: false
```
Comment (jeancochrane) [inline .github/workflows/lint.yaml:33]: The Markdown linter doesn't understand how to deal with the Jinja templating in dbt Markdown files, and as far as I can tell there's no way to exclude files from markdownlint via the superlinter config, so I went ahead and turned it off.

**Review by wrridgeway — APPROVED** [review]
Thank you @jeancochrane !

**Pull request merged**